### PR TITLE
Migrate job specific methods

### PR DESCRIPTION
**Pull request by smarlowucf**
### What does this PR do? Why are we making this change?

Migrate all methods specific to handling jobs from base mash service class to the pipeline, obs and credentials service classes. Also, move any related attrs.

### How will these changes be tested?

Unit tests and local integration testing.

### CODE REVIEW

File: mash/services/credentials/service.py [hunk 1]
```diff
@@ -373,13 +389,52 @@ def delete_account(self, message):
             requesting_user, account_name, cloud
         )
 
+    def remove_file(self, config_file):
+        """
+        Remove file from disk if it exists.
+        """
+        try:
+            os.remove(config_file)
+        except Exception:
```
Comment (rjschwei) [inline mash/services/credentials/service.py]: I think the specific exception should be caught here, i.e. `except FileNotFoundError:`

File: mash/services/obs/service.py [hunk 2]
```diff
@@ -237,3 +260,29 @@ def _start_job(self, job):
             'ok': True,
             'message': 'Job started'
         }
+
+    def persist_job_config(self, config):
```
Comment (rjschwei) [inline mash/services/obs/service.py]: Maybe worth to push this into a utility function to be re-used. This is the second implementation of the "persist" code. Not a "must change" item

File: mash/services/pipeline_service.py [hunk 3]
```diff
@@ -558,14 +578,68 @@ def publish_credentials_request(self, job_id):
...
+    def remove_file(self, config_file):
+        """
+        Remove file from disk if it exists.
+        """
+        try:
+            os.remove(config_file)
+        except Exception:
+            pass
+
+    def persist_job_config(self, config):
```
Comment (rjschwei) [inline mash/services/pipeline_service.py]: Number 3

File: mash/services/credentials/service.py (hunk 1 above)
Comment (smarlowucf) [reply mash/services/credentials/service.py]: Fair, actually I thought we had a context manager for ignoring things that are just set to `pass`. Although, we still want to catch general exceptions and maybe those get logged. We don't want an un-handled exception crashing the service.

Comment (rjschwei) [reply mash/services/credentials/service.py]: Well we really only want to ignore the "file not there" problem. If there is another exception that could point to a larger issues. The other exception would be that we cannot remove the file which might imply that the permissions have changed from under us and in that case we probably shouldn't continue as the system would be in a wacky state.

File: mash/services/obs/service.py (hunk 2 above)
Comment (smarlowucf) [reply mash/services/obs/service.py]: There is some duplication with these changes. The question is, is the duplication or being tied to a shared util module a bigger issue? As we decided that certain services in MASH would eventually be independently scalable we have to consider where it makes sense to use shared utility methods. An method that is shared by different services (not packaged together) will need to be it's own package which adds complexity.

E.g. if had two packages one for core mash (w/ credentials) and one for pipeline then we need a separate utility package that would hold the persist_job_config that is required by both.

**Comment by smarlowucf** [comment]
Rebased to fix conflicts.

File: mash/services/obs/service.py (hunk 2 above)
Comment (rjschwei) [reply mash/services/obs/service.py]: Sorry for the delayed answer. Yes, dependency management comes with it's own set of problems and yes it does add a certain level of complexity. However, the dependency resolution is a well understood and generally resolved problem and therefore I do consider this as only a marginal increase in complexity.

W.r.t. code duplication I can only speak from experience in that code that  is "reused" via duplication or cut and paste in inevitably drifts apart and that is generally a much bigger problem then the added complexity of managing dependencies for sharing code.

Thanks for creating the utility functions.

If we get to the point where mash needs to separate into different packages, Python and or RPM, then we'll deal with the "utils" packaging problem.

Comment (smarlowucf) [reply mash/services/obs/service.py]: Ack :+1:

**Pull request merged**